src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -82,33 +84,61 @@ const getRelationParts = (val: unknown): string[] => {
-  // look explicitly for an app named 'wds-${app.workspaceId}'. If found, use it, even if it isn't running
-  // this handles the case where the user has explicitly shut down the app
+// Invokes logic to determine a URL for WDS
+// If WDS is not running, a URL will not be present -- in some cases, this function may invoke
+// a new call to Leo to instantiate a WDS being available, thus having a valid URL
+export const resolveWdsUrl = (apps, workspaceId) => {
+  // WDS looks for Kubernetes deployment statuses (such as RUNNING or PROVISIONING), expressed by Leo
+  // See here for specific enumerations -- https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala
+  // look explicitly for a RUNNING app named 'wds-${app.workspaceId}' -- if WDS is healthy and running, there should only be one app RUNNNING
   const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}` && app.status === 'RUNNING')
   if (namedApp.length === 1) {
     return namedApp[0].proxyUrls.wds
   }
-  // if we didn't find the expected app 'cbas-wds-default', go hunting:
-  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.status === 'RUNNING')
+
+  // if we didn't find the expected app 'wds-${app.workspaceId}' running...
+  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}`)
+  // ...nothing has launched yet, bring WDS to life!
   if (candidates.length === 0) {
-    // no app deployed yet
+    // !getConfig().isProd is temporary until WDS is ready for production launch in Feb 2023
+    if (!getConfig().isProd) {
+      withErrorReporting('An error occurred when creating your data tables. Please reach out to support@terra.bio', async () => {
+        await Ajax().Apps.createAppV2(`wds-${workspaceId}`, workspaceId)
+      })
+    }
+    return ''
+  }
+
+  // WDS is being created in k8s (takes a few minutes)
+  if (candidates.length === 1 && candidates[0].status === 'PROVISIONING') {
     return ''
   }
-  if (candidates.length > 1) {
-    // multiple apps found; use the earliest-created one
+
+  // If we reach this logic, we have more than one Leo app with the associated workspace Id...
+  const healthyStatuses = ['RUNNING', 'PROVISIONING']
+  const healthyCandidates = candidates.filter(app => healthyStatuses.includes(app.status))
+  if (healthyCandidates > 0) {
+    // Evaluate the earliest-created WDS app
     candidates.sort((a, b) => a.auditInfo.createdDate - b.auditInfo.createdDate)
+    if (candidates[0].status === 'RUNNING') {
+      return candidates[0].proxyUrls.wds
+    }
+    if (candidates[0].status === 'PROVISIONING') {
+      return ''
+    }
+    // TODO: AJ-790: How do we feel if we reach this point...
+    return 'ERROR'
   }
-  return candidates[0].proxyUrls.wds
+  // TODO: AJ-790: How do we feel if we reach this point...
+  return 'ERROR'
```

Comment:
if we return `ERROR` here, the check in Data.js will actually return true: https://github.com/DataBiosphere/terra-ui/blob/659f8b769140a9d47214443adb29ead28076bc22/src/pages/workspaces/workspace/Data.js#L577 and we'll use the string `ERROR` as the base url for WDS requests … which will of course cause errorrs

Reply:
I think we might need to edit what is presented to the end user then -- e.g. if we return an empty string, the spinning wheel will present itself as if `PROVISIONING` might be the case.

Let me know your thoughts here -- we could build out the logic in the UI for that spinning wheel option?

Cc @bethsheets

Reply:
for the scope of this PR, I suggest we do exactly the same as if the auto-deploy upon new workspace creation resulted in error or other non-provisioning/non-running status. I realize the spinning wheel is not ideal, but I suggest we tackle that in a future PR and not block this one on that decision.

Reply:
specifically, I think we'd handle all those non-provisioning/non-running statuses in AJ-783

Reply:
Sounds good -- I've made every scenario (other than when `RUNNING` is the case) return an empty string for now